pkg/expr/nodes.go

```diff
@@ -236,6 +236,41 @@ func (dn *DSNode) Execute(ctx context.Context, vars mathexp.Vars, s *Service) (m
 			return mathexp.Results{}, QueryError{RefID: refID, Err: qr.Error}
 		}
 
+		dataSource := dn.datasource.Type
+		if dataSource == "prometheus" {
```

Comment:
Maybe make this and the block below it extracted into funcs

Reply:
What is special about a vector result that needs special handling?

Reply:
SSE has similar types to Prometheus, Matrix/Vector (Except TimeSeries / Number in the case of SSE). The number type can be alerted on directly, whereas series need to be reduced. So without this indicator, SSE wouldn't know if a response is a time series of length 1, or a number response.